beta/src/pages/learn/adjusting-effect-dependencies.md

```diff
@@ -0,0 +1,1345 @@
+
+Effects re-run every time a value you read inside has changed. This makes sense for most code inside Effects: for example, [when you started using `roomId` inside the Effect,](#every-reactive-value-becomes-a-dependency) it made sense to reconnect if the `roomId` has changed. However, code inside `handleMessage` should behave like an event handler: [editing it shouldn't affect when the code runs.](#event-handlers-run-on-specific-interactions) **Extract `handleMessage` into an *Event function* out of your Effect with the [`useEvent`](/api/useevent) Hook:**
+
+```js {1,7-12}
+import { useState, useEffect, useEvent } from 'react';
+
+function ChatRoom({ roomId }) {
+  const [messages, setMessages] = useState([]);
+  const [isMuted, setIsMuted] = useState(false);
+
+  const handleMessage = useEvent(receivedMessage => {
+    setMessages(msgs => [...msgs, receivedMessage]);
+    if (!isMuted) {
+      playSound();
+    }
+  });
+
+  useEffect(() => {
+    const connection = createConnection(roomId);
+    connection.on('message', handleMessage);
+    connection.connect();
+    return () => {
+      connection.on('message', handleMessage);
+      connection.disconnect();
+    };
+  }, [roomId]); // ✅ All dependencies declared
+
+  function handleMuteClick() {
+    setIsMuted(!isMuted);
+  }
+
+  // ...
+```
+
+Code inside Effects is reactive. It re-runs when the values inside change. If you don't want a part of your Effect's code to be reactive, extract it to an Event function like `handleMessage` above, and call that Event function from your Effect. **Event functions are omitted from dependencies. Editing them won't make any code run more often.**
+
+#### Wrapping callbacks into Event functions {/*wrapping-callbacks-into-event-functions*/}
+
+> **Code examples in this section don't work yet!** They rely on a [proposed `useEvent` API](https://github.com/reactjs/rfcs/pull/220).
+
+Sometimes, the function you want to call may be passed from a parent component. If you don't want your Effect to "react" to changes to that function, wrap it into an Event function next to where you declare your Effect:
+
+```js {2,6,9}
+function ChatRoom({ roomId, onMessage }) {
+  const handleMessage = useEvent(onMessage);
+
+  useEffect(() => {
+    const connection = createConnection(roomId);
+    connection.on('message', handleMessage);
+    connection.connect();
+    return () => {
+      connection.on('message', handleMessage);
+      connection.disconnect();
+    };
+  }, [roomId]); // ✅ All dependencies declared
+
+  // ...
+```
+
+If you tried to call `onMessage` directly from your Effect, you would have to include `onMessage` into your dependencies. This would cause your Effect to re-run whenever the parent component re-renders. However, `handleMessage` that you got from the `useEvent(onMessage)` declaration is an Event function. This means that you don't need to specify it in the Effect dependencies. Now it can never make your Effect re-run accidentally.
```

Comment:
add something about even though the effect doesn't rerun, it'll call the latest value of onMessage? (latest committed value)